woocommerce/wc-admin
# Add v4 orders endpoint

Fixes #950 

Adds the new endpoint so we can use new filters and return data from lookup tables.

### Screenshots
<img width="563" alt="screen shot 2019-01-09 at 2 32 29 pm" src="https://user-images.githubusercontent.com/10561050/50881066-728c9880-141b-11e9-891b-96af6018282a.png">

### Detailed test instructions:

1.  Make sure you have coupons and products associated with orders in your lookup tables.
2.  Rebuild your reports to ensure statuses are in the lookup tables.
3.  Send a request to `wp-json/wc/v4/reports/orders` and test the following params **(note the v4 namespace)**:
`before (date)`
`after (date)`
`page (int)`
`per_page (int)`
`orderby (date, num_items_sold, or net_total)`
`order (asc or desc)`
`status_is (string - status type without 'wc-' prefix)`
`status_is_not (string - status type without 'wc-' prefix)`
`product_includes (int or array)`
`product_excludes (int or array)`
`coupon_includes (int or array)`
`coupon_excludes (int or array)`
`customer_type (string 'returning' or 'new')`
`extended_info (bool)`
4.  Run phpunit tests to make sure everything is ✅
